core/src/main/scala/org/http4s/blaze/channel/nio2/ClientChannelFactory.scala

```diff
@@ -30,18 +41,33 @@ final class ClientChannelFactory(
 
     try {
       val ch = AsynchronousSocketChannel.open(group.orNull)
+
+      val scheduledTimeout = connectingTimeout.map { t =>
+        val onTimeout = new Runnable {
+          override def run(): Unit = {
+            val finishedWithTimeout = p.tryFailure(new SocketTimeoutException())
```

Comment:
Would it be useful to include the remote address and maybe the timeout in a message?